mkosi/__init__.py

```diff
@@ -8059,14 +8058,8 @@ def find_address(args: MkosiArgs) -> Tuple[str, str]:
     die("Container/VM address not found")
 
 
-def run_command_image(args: MkosiArgs, commands: Sequence[str], timeout: int, check: bool, stdout: _FILE = sys.stdout, stderr: _FILE = sys.stderr) -> CompletedProcess:
-    if args.verb == Verb.qemu:
-        return run_ssh(args, commands, check, stdout, stderr, timeout)
-    elif args.verb == Verb.boot:
-        cmdline = ["systemd-run", "--quiet", "--wait", "--pipe", "-M", machine_name(args), "/usr/bin/env", *commands]
-        return run(cmdline, check=check, stdout=stdout, stderr=stderr, text=True, timeout=timeout)
-    else:
-        return run(run_shell_cmdline(args, pipe=True, commands=commands), check=check, stdout=stdout, stderr=stderr, text=True, timeout=timeout)
+def run_systemd_cmdline(args: MkosiArgs, commands: Sequence[str]) -> Sequence[str]:
+    return ["systemd-run", "--quiet", "--wait", "--pipe", "-M", machine_name(args), "/usr/bin/env", *commands]
```

Comment:
I never noticed this `/usr/bin/env` here. I'm not quite sure I understand what it's doing?

Reply:
systemd-run creates a transient unit, and ExecStart= expected absolute paths until some recent systemd version, so we prefix commands with env to make sure binaries are looked up in the path.